src/draw/DefaultDrawer.js

```diff
@@ -106,466 +160,464 @@ class DefaultDrawer {
+        .attr('preserveAspectRatio', 'xMinYMin meet')
+        .style('font', '10px sans-serif');
+      this.svg.append('g')
+        .attr('class', 'container')
+        .attr('height', '100%')
+        .attr('width', '100%')
+        .attr('x', 0)
+        .attr('y', 0);
+    } else {
+      this.svg = d3.select(`#${this.rootId}`)
+        .select('svg');
+    }
   }
 
   /**
-   * Set actions on component.
-   * @param {Component[]} components - Array containing components.
+   * Draws all Components and ComponentLinks in the parentId Element.
+   * @param {String} [rootId] - Id of the container where you want to draw.
+   * @return {Boolean} Returns true if normally execute.
    */
-  setComponentAction(components) {
-    this.__drag(components);
-    this.__dropToRoot(components);
-    components.forEach((component) => {
-      if (component.definition.isContainer) {
-        this.__dropToContainer(components, component);
-      }
-    });
+  draw(rootId) {
+    this.rootId = rootId;
+    this.createRenderingContext();
 
-    this.interact('#action-menu .trash').unset();
-    this.interact('#action-menu .trash').on('click', () => {
-      this.__deleteComponent(this.actions.selection.current);
-    });
+    this.__unselectComponent();
 
-    this.interact('#action-menu .edit').unset();
-    this.interact('#action-menu .edit').on('click', () => {
-      this.__editComponent(this.actions.selection.current);
-    });
+    this.drawComponents();
 
-    this.interact('#action-menu .link').unset();
-    this.interact('#action-menu .link').on('click', () => {
-      const allowedTarget = this.pluginData.definitions.links.filter((def) => def.sourceRef
-          === this.d3.select(`#${this.actions.selection.current}`).datum().definition.type);
+    this.drawLinks();
+    this.initializeActionMenu();
 
-      this.actions.link.targetList = allowedTarget.map((def) => def.targetRef);
+    this.setViewPortAction(d3.select(`#${this.rootId}`));
 
-      this.d3.selectAll('.component')
-        .style('opacity', (data) => {
-          if (this.actions.link.targetList.length > 0) {
-            return this.actions.link.targetList.includes(data.definition.type) ? 1 : 0.5;
-          }
-          return 1;
-        });
-      this.interact('.component').on(
-        'click',
-        (event) => {
-          const targetType = this.d3.select(`#${event.currentTarget.id}`)
-            .datum()
-            .definition
-            .type;
-
-          if (!this.actions.link.targetList.includes(targetType)) {
-            this.__unsetAddLink();
-          }
-        },
-        { once: true },
-      );
-      this.__addLink(allowedTarget);
-    });
+    return true;
   }
 
   /**
-   * Delete component
-   * @private
-   */
-  __deleteComponent(componentId) {
-    if (this.events.DeleteEvent) {
-      this.events.DeleteEvent.next({
-        id: componentId,
-      });
+   *
+   * @param {PointerEvent} event - The click event
```

Comment:
please check all your `@param` and another tag to have `.` at the end of line

Reply:
done